app/models/debt_transfer.rb

```diff
@@ -71,7 +71,8 @@ class DebtTransfer < Ekylibre::Record::Base
                     elsif debt_transfer_affair.is_a?(SaleAffair) && affair.is_a?(PurchaseAffair)
                       :purchase_regularization
                     else
-                      raise 'Cannot run a debt transfer with homogeneous affairs'
```

Comment:
This simply can't work. The nature of `DebtTransfer`s is enumerized to only accept `:sale_regularization` or `:purchase_regularization` as value. This would directly raise a validation error when run.
The fix must be implemented at a higher scope, the users should simply not be able to try and create a debt transfer between two affairs of the same kind.